modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/Processing.scala

```diff
@@ -131,8 +131,8 @@ object Processing {
     }
 
   /** Parse raw bytes into Event using analytics sdk */
-  private def parseBytes[F[_]: Sync](badProcessor: BadRowProcessor): Pipe[F, TokenedEvents, ParseResult] =
-    _.evalMap { case TokenedEvents(chunk, token) =>
+  private def parseBytes[F[_]: Async](badProcessor: BadRowProcessor, parallelism: Int): Pipe[F, TokenedEvents, ParseResult] =
+    _.parEvalMapUnordered(parallelism) { case TokenedEvents(chunk, token) =>
```

Comment:
This must not be unordered, because it means we could checkpoint batches out of order.  For Kinesis and Kafka we must keep the order of batches.

See equivalent line in Snowflake loader: https://github.com/snowplow-incubator/snowflake-loader/blob/0.3.0/modules/core/src/main/scala/com.snowplowanalytics.snowplow.snowflake/processing/Processing.scala#L147

Reply:
Oh indeed, very good point, I forgot about it because I was working with Pubsub